packages/builder/src/stores/builder/views.ts

```
@@ -0,0 +1,94 @@
+import { DerivedBudiStore } from "../BudiStore"
+import { tables } from "./tables"
+import { API } from "@/api"
+import { View } from "@budibase/types"
+import { helpers } from "@budibase/shared-core"
+import { derived, Writable } from "svelte/store"
+
+interface BuilderViewStore {
+  selectedViewName: string | null
+}
+
+interface DerivedViewStore extends BuilderViewStore {
+  list: View[]
+  selected?: View
+}
+
+export class ViewsStore extends DerivedBudiStore<
+  BuilderViewStore,
+  DerivedViewStore
+> {
+  constructor() {
+    const makeDerivedStore = (store: Writable<BuilderViewStore>) => {
+      return derived([store, tables], ([$store, $tables]): DerivedViewStore => {
+        let list: View[] = []
+        $tables.list?.forEach(table => {
+          const views = Object.values(table?.views || {}).filter(
+            (view): view is View => !helpers.views.isV2(view)
+          )
+          list = list.concat(views)
+        })
+        return {
+          selectedViewName: $store.selectedViewName,
+          list,
+          selected: list.find(view => view.name === $store.selectedViewName),
+        }
+      })
+    }
+
+    super(
+      {
+        selectedViewName: null,
+      },
+      makeDerivedStore
+    )
+
+    this.select = this.select.bind(this)
+  }
+
+  select = (name: string) => {
+    this.store.update(state => ({
+      ...state,
+      selectedViewName: name,
+    }))
+  }
+
+  delete = async (view: View) => {
+    if (!view.name) {
+      return
```

Is this an expected use case? Should we throw an error?